### DISCUSSION TITLE
Umi & Bigfish 想法征集 03

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10639] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊、云谦），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by sorrycc** [comment]
1、关于如何从 tsx 中解析出 import？感觉有几个解法，1）FORK 一份 es-module-lexer 做 tsx 的支持，2）[ast-grep](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6InR5cGVzY3JpcHQiLCJxdWVyeSI6ImNvbnNvbGUubG9nKCRNQVRDSCkiLCJjb25maWciOiIjIENvbmZpZ3VyZSBSdWxlIGluIFlBTUxcbnJ1bGU6XG4gIGFsbDpcbiAgICAtIGtpbmQ6IGltcG9ydF9zcGVjaWZpZXJcbiAgICAtIHJlZ2V4OiB0eXBlXFxzXG5jb25zdHJhaW50czpcbiAgIyBNRVRBX1ZBUjogcGF0dGVybiIsInNvdXJjZSI6ImltcG9ydCB7dHlwZSBBLCBCLCBDLCB0eXBlIEREREREfSBmcm9tICdiJ1xuXG52YXIgYTogQSA9IDEyMyJ9)
2、参考 Bigfish Checker 做 Umi Checker，用于发布前跑开源三方应用的构建，比如 antd 官网
3、Bigfish 交互式课程和 Dumi 应该都会用到 [Button | Mantine](https://mantine.dev/core/button/) 这动图修改 props 的能力
4、学习下 Next.js 的错误提示框

**Reply by stormslowly** [reply]
感觉 ast-grep 会更容易一点

**Reply by fz6m** [reply]
`ast-grep` 不支持 jsx 、ts 最新语法，也需要先 esbuild 去 type 。

**Reply by stormslowly** [reply]
> `ast-grep` 不支持 jsx 、ts 最新语法，也需要先 esbuild 去 type 。

sad 😭

**Reply by HerringtonDarkholme** [reply]
我可以帮忙升级tree-sitter让ast-grep支持最新语法。目前`satisfies`等语法已经支持了。还有不支持的可以留言给我案例。

**Comment by xiaohuoni** [comment]
1、关于旧的 [umi ui](https://github.com/umijs/umi-ui)，尝试升级到 umi@4 ，能看到页面，但是很多功能仍然有问题。感觉代码有点老旧了，我放弃了，@chiaweilee 还在努力。
2、关于新的 [umi ui](https://github.com/umijs/umi/tree/master/ui)，现在我已知的需求，没有我要的东西。
3、关于我自己想要的 umi ui，我想要的是一个支持自定义扩展的架子，能够让一些“内部组件库”搭建属于自己的 Umi 开发者工具，当前大部分的需求都是我和 chiaweilee 的需求。详见 [mongchhi](https://github.com/umijs/mongchhi)
4、关于 @chiaweilee 想要的 umi ui ，https://www.yuque.com/g/jiajiada/wf9n0x/agmuifuoq4ew20yl/collaborator/join?token=RqSetlnPZCDAMOq6&goto=%2Fjiajiada%2Fwf9n0x%2Fagmuifuoq4ew20yl%2Fedit# 
5、[进行中] `api.describe({ config: { schema() { } } })` 增加 zod 支持。之前被其他事情冲淡优先级，最近 @xierenyuan 也想要。

**Reply by sorrycc** [reply]
1、https://www.yuque.com/jiajiada/wf9n0x/agmuifuoq4ew20yl 这个打不开
2、umi ui 会支持扩展的

**Comment by stormslowly** [comment]
* 调研 Next-js， telemetry, 目标: 产出  umi Telemetry RFC
  * 调研 sandpack 和 stackblitz ，目标：产出一个可嵌入 umi 命令行实例的组件。

这周看的东西少先水两个，下周补上

**Reply by sorrycc** [reply]
1、sandpack 在内网 bigfish/bigfish-course 仓库已封装了一个
2、stackblitz 需要注意的是，他依赖外网服务，不支持私域的包

**Comment by hanzebang** [comment]
1. logger 支持自定义路径
  - 需求背景：比如 doctor 使用 @umijs/utils 的 logger 时，打的日志和应用日志是混在一起的，目前日志路径是写死的（指 `node_modules/.cache/logger/umi.log` 日志）
2. logger info 等级别日志支持隐藏
  - 需求背景：doctor 跑 service 时会输出很多 service 的日志（doctor 可能并不关心），不太好看。。

**Reply by xiaohuoni** [reply]
关于2: doctor 读文件的时候，自己按需求过滤是不是更合理？感觉 umi ui 也可能会展示这个 log。

**Reply by hanzebang** [reply]
> 关于2: doctor 读文件的时候，自己按需求过滤是不是更合理？感觉 umi ui 也可能会展示这个 log。

「自己按需过滤」具体是指？

目前 doctor 跑 service 时，会不可避免的先打印以下几行日志：
```bash
info  - [你知道吗？] xxxx。
info  - generate files
info  - Preparing...
info  - [icons] generate icons local:home
```

**Reply by sorrycc** [reply]
1、问题 1 记录了
2、你知道吗可以通过环境变量关闭，其他感觉是正常日志

**Comment by PeachScript** [comment]
1. `onBuildComplete` 钩子在 error 情况下部分调用失效的问题，已有 RFC 可直接讨论解决方案：https://github.com/umijs/umi/pull/10632
2. Umi/Bigfish 官网迁移到 dumi，获得全文搜索技能
3. ssr 的 dev middleware 与原生 dev 不一致，比如 modifyHTML 无法生效之类的

**Reply by sorrycc** [reply]
问题 3 已记录。
